### PR TITLE
[MRG] Document files module

### PR DESCRIPTION
**Pull request by mbeyeler**
Add documentation and testing of the `files` module:

- [X] Document functions
- [X] Test functions
- [X] Trigger `ImportError`
- [X] Improve coverage

### CODE REVIEW

File: pulse2percept/utils.py [hunk 1]
```diff
@@ -13,8 +16,27 @@
     has_jit = False
 
 
+def deprecated(func):
```
Comment (arokem) [inline pulse2percept/utils.py]: Maybe add a key-word argument for "alt_func", which, if defined, will trigger a block that also prints a `"Use {} instead".format(alt_func.__name__)`?

File: pulse2percept/tests/test_files.py [hunk 2]
```diff
@@ -1,5 +1,57 @@
...
+try:
+    from unittest import mock
+except ImportError:
+    import mock
+
+from pulse2percept import files
+
+
+def test_savemoviefiles():
+    # This function is deprecated
```
Comment (arokem) [inline pulse2percept/tests/test_files.py]: Do you want to test that the warning is raised?

**Comment by mbeyeler** [comment]
The nested decorator turned out to be a bit of a nightmare. [This advice](http://stackoverflow.com/questions/35572663/using-python-decorator-with-or-without-parentheses) turned out to be the best one (also [this](https://github.com/pytest-dev/pytest/blob/2e04771893f17b571e6ab48f6884a6a70cde8fc8/_pytest/python.py#L132)).

It works fine if you do `@deprecated(arg=p2p.utils.TimeSeries)` or something. Problem is if you do `@deprecated`, without passing `arg`, the current function object gets assigned to `arg`, still. So there's no way to differentiate whether `arg` is the current function object or the alternative function object.
Solution: Don't pass the object of the alternative function, use a string instead.

Ready to merge.

File: pulse2percept/files.py [hunk 3]
```diff
@@ -1,31 +1,71 @@
...
+    Parameters
+    ----------
+    filename : str
+        File name of .avi movie to be produced
+    movie : array
+        A 3-D NumPy array containing the data, such as the `data` field of
+        a utils.TimeSeries object
+    rate : float, optional
+        Frame rate. Default: 30 Hz
+    """
```
Comment (arokem) [inline pulse2percept/files.py]: Please add the `.. deprecated:: 0.2` tag here.

File: pulse2percept/utils.py [hunk 4]
```diff
@@ -13,8 +14,55 @@
...
+        # Direct decoration
+        return wrap(arg)
+    else:
+        # Alternative function given
+        alt_func = arg
+        return wrap
+
+
 class Parameters(object):
+    """Container to wrap a MATLAB array in a Python dict"""
```
Comment (arokem) [inline pulse2percept/utils.py]: Add a note that this is going to be deprecated in 0.2

**Comment by mbeyeler** [comment]
Great point about the `arg`. I removed it from kwargs and made it an arg instead. I also added the deprecated flags you mentioned.

File: pulse2percept/utils.py [hunk 5]
```diff
@@ -42,7 +42,7 @@ def wrap(func):
         """
         e_s = "Call to deprecated function %s." % func.__name__
         if alt_func:
-            e_s += "Use %s instead." % alt_func
+            e_s += " Use %s instead." % alt_func
```
Comment (arokem) [inline pulse2percept/utils.py]: Sorry - I think I missed this before. Where does `alt_func` come from? 

It doesn't seem to be defined anywhere. I must be missing something

Comment (mbeyeler) [inline pulse2percept/utils.py]: It is set outside `wrap()` but inside `deprecated()` (starting l.57), in the case where `arg` is not callable:
```
alt_func = None
if callable(arg):
    # Direct decoration
    return wrap(arg)
else:
    # Alternative function given
    alt_func = arg
    return wrap
```

**Comment by arokem** [comment]
I knew I was missing something.

**Pull request merged**